Project reorganization for Unity Package Manager (UPM) support (#5)
pusher/pusher-websocket-unity: README.md

- All Unity 2019.2.x

## Getting Started
1. [Create a Pusher Channels app](#1-create-a-pusher-channels-app)
2. [Install](#2-install) the **Pusher Channels Unity Client Library** via `unitypackage` or via Unity Package Manager (UPM)
3. [Add PusherManager and run the game](#3-add-pushermanager-and-run-the-game)

### 1 Create a Pusher Channels app
1.1 - Create a Pusher Channels app at https://pusher.com/channels

### 2 Install
#### 2.1 Install it via unitypackage

> 2.1.1 - Download the latest `PusherWebsocketUnity-1.0.2+190808.unitypackage` from this repo's [Releases](/../../releases)<br>

Reviewer: ```suggestion
2.1.1 - Download the latest `PusherWebsocketUnity-1.0.2+190808.unitypackage` from [releases](/../../releases)<br>
```

2.2.1 - [if your Unity version is **2018.x.x**] Make sure that under `Edit -> Project Settings -> Player` the `Configuration -> Scripting Runtime Version` is set to **.NET 4.x Equivalent**.<br>
2.2.2 - Open the `Packages/manifest.json` with your favourite editor and add the following in your `dependencies` (make sure to respect JSON commas):
```
{
 "dependencies": {
  ...
  "com.pusher.pusherwebsocketunity": "https://github.com/pusher/pusher-websocket-unity.git#1.0.2+190808"
 }
}
```
2.2.3 - Now Unity should auto resolve dependencies and fetch the newly defined package.

### 3 Add PusherManager and run the game
3.1 - Copy the sample [`PusherManager.cs`](BaseProject/Assets/PusherManager.cs) into **your project's Assets folder** and add your keys as values for `private const string APP_KEY` and `private const string APP_CLUSTER` obtained when you created the Pusher Channels app in the dashboard.<br>
3.2 - Create a new GameObject, by going on `GameObject -> Create Empty`. Drag the `PusherManager.cs` script onto the GameObject Inspector to set it as a script for the object.<br>
3.3 - Save and click Play to start the game in Unity.<br>

> 3.4 - Verify that in the Console tab the following is logged: `Connection state changed`, `Connected`, `Subscribed`.<br>

Reviewer: ```suggestion
3.4 - Verify that `Connection state changed`, `Connected`, `Subscribed` is logged in the *Console* tab.<br>
```

3.5 - You can now customize the channel name (by default is `"my-channel"`) and events to bind to (by default is `"my-event"`) in the `PusherManager.cs` script.

## Update this package
### Update the version of .unitypackage

> To update the version of the `.unitypackage` you can just delete the `PusherWebsocketUnity` folder under your `Assets`, and repeat the [Install steps](#2-install).

Reviewer: ```suggestion
To update the version of the `.unitypackage` you can delete the `Assets/PusherWebsocketUnity` directory and repeat the [install steps](#2-install).
```

### Update the version of Unity Package Manager (UPM) package

> To update the version of `com.pusher.pusherwebsocketunity` you just need to check the available versions in [Releases](/../../releases) and change the package version in the `Packages/manifest.json`, like so:

Reviewer: ```suggestion
To update the version of `com.pusher.pusherwebsocketunity` you just need to check the available versions in [releases](/../../releases) and change the package version in the `Packages/manifest.json`, like so:
```
